platform/ios/src/MGLMapView.mm

```diff
@@ -3550,6 +3555,10 @@ - (MGLAnnotationTag)annotationTagAtPoint:(CGPoint)point persistingResults:(BOOL)
         auto end = std::remove_if(nearbyAnnotations.begin(), nearbyAnnotations.end(),
                                   [&](const MGLAnnotationTag annotationTag)
         {
+            if (annotationTag == MGLAnnotationTagNotFound) {
```

Comment:
Also, this is the assertion we hit in debug builds if annotations are removed at almost the same time and just barely before the user tries to select them (part of the repro from https://github.com/mapbox/mapbox-gl-native/issues/8163#issuecomment-291695262). 

The check below recently added in https://github.com/mapbox/mapbox-gl-native/pull/8637 to make sure the annotation is not nil protects release builds from crashing at the call to `std::map::at` also just below.